js/data/bucket.js

```diff
@@ -150,34 +147,42 @@ Bucket.prototype.createArrays = function() {
         var programInterface = this.programInterfaces[programName];
 
         if (programInterface.vertexBuffer) {
-            var vertexAddMethodName = this.getAddMethodName(programName, 'vertex');
             var vertexBufferName = this.getBufferName(programName, 'vertex');
 
             var VertexArrayType = new StructArrayType({
-                members: this.attributes[programName].enabled,
+                members: this.attributes[programName].coreAttributes,
```

Comment:
good idea, done 7852535e6e80353f761671f9c4c5263a68ec99de. The paint/layout categorization is surviving dds better than I thought it would